develop/how-to-submit-patches/index.md

```diff
@@ -0,0 +1,56 @@
+
+Remember that if you have any questions about the steps here, you can always post them to wx-dev mailing list and we'll do our best to help you.
+
+## Changing wxWidgets
+
+ * **Discuss your changes**: Before making your changes, please post to the [developers mailing list](/support/mailing-lists/) describing what you plan to do. Unless your change is very simple, this may save a lot of time to everyone involved compared to making the change first and then having to change it later.
+
+ * **Follow our coding rules**: Please read the [wxWidgets coding standards](/develop/coding-guidelines/) and at least try to conform to them. In particular, please respect the indentation rules (4 spaces, no TABs) and code style (braces must be on their own line etc).
+
+ * **Unit tests**: Please update (or add) unit tests for any easily testable functionality aspect that you modify or add. Our test suite doesn't cover 100% of wxWidgets code yet but we'd really like it to and we prefer not to add new classes or functions without tests.
+
+ * **Documentation**: Last but not least, if your patch adds a new feature please include the patch to the documentation as well as undocumented feature is hardly useful to anyone but its author. The API is documented in `/interface` directory, in familiar and simple [Doxygen](http://www.doxygen.org/) format. If you don't do it, another developer would have to write the documentation and the patch won't be applied until he has time to do it.
+
+## Mechanics of patch submission
+
+If possible, i.e if you already use GitHub, please submit your patches via GitHub pull requests (PR) mechanism, i.e. fork [wxWidgets repository](https://github.com/wxWidgets/wxWidgets/), make a branch with your changes and submit PR to wx (master or 3.0). This has the advantage of building your patch in several different build configurations and rerunning the tests, so it's the preferred way of submitting your changes. The rest of this section only applies if you are unable or unwilling to use this mechanism.
+
+While we prefer merging pull requests, we still accept plain patches too. For those, please use the manual approach described here:
+
+ * **Use the latest version**: Please try to always make your patches against the latest version of wxWidgets.
+
+  In most cases, you should make a patch against the git master branch. If the bug is only in stable branch or it was already fixed in master, make the patch against the stable git branch. You can learn how to download the source from git [here](/develop/code-repository/).
+
+  If you cannot access git (for example when you are behind a firewall), please make the patch against latest release or, better, against fresh git snapshot (latest download: [master.zip](https://github.com/wxWidgets/wxWidgets/archive/master.zip)). 
```

Comment:
The GIT snapshot unfortunately does not include the code from GIT submodules, which many people find surprising. Not sure whether something can be done about it.

Reply:
Hopefully should be fine for making patches? In any case, I'm not sure what else to recommend here.